liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java

```diff
@@ -299,44 +288,169 @@ public ServerTask getServerTask() throws Exception {
-                        unhandledChange = true;
-                    }
-                }                
-                if (!allDifferences.isEmpty()) {
-                    log.info("Pom has been modified");
-                    if (isUsingBoost()) {
-                        log.info("Running boost:package");
-                        runBoostMojo("package", true);
+                build = mavenProjectBuilder.build(buildFile,
+                        session.getProjectBuildingRequest().setResolveDependencies(true));
+            } catch (ProjectBuildingException e) {
+                log.error("Could not parse pom.xml. " + e.getMessage());
+                log.debug(e);
+                return false;
+            }
+
+            // set the updated project in current session;
+            Plugin backupLibertyPlugin = getPlugin(LIBERTY_MAVEN_PLUGIN_GROUP_ID, LIBERTY_MAVEN_PLUGIN_ARTIFACT_ID);
+            MavenProject backupProject = project;
+            project = build.getProject();
+            session.setCurrentProject(project);
+            Plugin libertyPlugin = getPlugin(LIBERTY_MAVEN_PLUGIN_GROUP_ID, LIBERTY_MAVEN_PLUGIN_ARTIFACT_ID);
+
+            try {
+                // Monitoring liberty properties in the pom.xml
+                if (hasServerPropertyChanged(project, backupProject)) {
+                    restartServer = true;
+                }
+                if (!restartServer && hasServerVariableChanged(project, backupProject)) {
+                    createServer = true;
+                }
+
+                // monitoring Liberty plugin configuration changes in dev mode
+                if (!restartServer && !createServer) {
+                    Xpp3Dom config = ExecuteMojoUtil.getPluginGoalConfig(libertyPlugin, "create", log);
+                    Xpp3Dom currentConfig = ExecuteMojoUtil.getPluginGoalConfig(backupLibertyPlugin, "create", log);
+
+                    if (!Objects.equals(config.getChild("bootstrapProperties"),
+                            currentConfig.getChild("bootstrapProperties"))) {
+                        restartServer = true;
+                    } else if (!Objects.equals(config.getChild("bootstrapPropertiesFile"),
+                            currentConfig.getChild("bootstrapPropertiesFile"))) {
+                        restartServer = true;
+                    } else if (!Objects.equals(config.getChild("jvmOptions"),
+                            currentConfig.getChild("jvmOptions"))) {
+                        restartServer = true;
+                    } else if (!Objects.equals(config.getChild("jvmOptionsFile"),
+                            currentConfig.getChild("jvmOptionsFile"))) {
+                        restartServer = true;
+                    } else if (!Objects.equals(config.getChild("serverEnvFile"),
+                            currentConfig.getChild("serverEnvFile"))) {
```

Comment:
Do we need to worry about the aliases? e.g. serverEnv

Reply:
I am going to rewrite it, I don't really need to worry about it. see my commit